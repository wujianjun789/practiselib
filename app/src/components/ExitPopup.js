import React from 'react';
import PropTypes from 'prop-types';
import PanelFooter from './PanelFooter';

const ExitPopup = ({tips='', iconClass='', cancel, confirm}) => {
    return <div className="usr-exit">
        <div className='tips'>
            <div><span className={`icon ${iconClass}`}></span></div>
            {tips}
        </div>
        <div className="btn-toolbar">
            <div className='btn-group'><button className="btn btn-default" onClick={cancel}>Cancel</button></div>
            <div className='btn-group'><button className="btn btn-primary" onClick={confirm}>Confirm</button></div>
        </div>
    </div>
}

ExitPopup.prototype = {
    tips: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.node
    ]).isRequired,
    iconClass: PropTypes.string.isRequired,
    cancel: PropTypes.func.isRequired,
    confirm: PropTypes.func.isRequired
}

export default ExitPopup;