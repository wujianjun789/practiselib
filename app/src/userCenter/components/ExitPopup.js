import React from 'react';

const ExitPopup = ({cancel, confirm}) => {
    return <div className="usr-exit">
        <div className='tips'>
            <div><span className="icon icon-popup-exit"></span></div>
            是否退出？
        </div>
        <div className="btn-toolbar">
            <div className='btn-group'><button className="btn btn-default" onClick={cancel}>Cancel</button></div>
            <div className='btn-group'><button className="btn btn-primary" onClick={confirm}>Confirm</button></div>
        </div>
    </div>
}

export default ExitPopup;