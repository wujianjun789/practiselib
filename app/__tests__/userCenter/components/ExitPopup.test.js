import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import ExitPopup from '../../../src/components/ExitPopup';

describe('<ExitPopup />', () => {
    it('default render', () => {
        const click = jest.fn();
        const cmp = shallow(<ExitPopup cancel={click} confirm={click}/>);

        let icon = cmp.find('.icon.icon-popup-exit');
        expect(icon.length).toBe(1);

        let txt = cmp.find('.tips');
        expect(txt.text()).toBe('是否退出？');

        let btnCancel = cmp.find('.btn.btn-default');
        expect(btnCancel.text()).toBe('Cancel');

        let btnConfirm = cmp.find('.btn.btn-primary');
        expect(btnConfirm.text()).toBe('Confirm');
    });

    it('click simulate', () => {
        const click = jest.fn();
        const cmp = shallow(<ExitPopup cancel={click} confirm={click}/>);

        let btnCancel = cmp.find('.btn.btn-default');
        btnCancel.simulate('click');
        expect(click).toHaveBeenCalledTimes(1);

        let btnConfirm = cmp.find('.btn.btn-primary');
        btnConfirm.simulate('click');
        expect(click).toHaveBeenCalledTimes(2);
    });

    it('snapshot', () => {
        const click = jest.fn();
        const cmp = renderer.create(<ExitPopup cancel={click} confirm={click}/>);
        let tree = cmp.toJSON();
        expect(tree).toMatchSnapshot();
    });
});