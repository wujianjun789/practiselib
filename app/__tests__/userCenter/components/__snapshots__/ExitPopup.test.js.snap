exports[`<ExitPopup /> snapshot 1`] = `
<div
  className="usr-exit">
  <div
    className="tips">
    <div>
      <span
        className="icon icon-popup-exit" />
    </div>
    是否退出？
  </div>
  <div
    className="btn-toolbar">
    <div
      className="btn-group">
      <button
        className="btn btn-default"
        onClick={[Function]}>
        Cancel
      </button>
    </div>
    <div
      className="btn-group">
      <button
        className="btn btn-primary"
        onClick={[Function]}>
        Confirm
      </button>
    </div>
  </div>
</div>
`;
